python/huggingfaceserver/huggingfaceserver/encoder_model.py

```diff
@@ -242,7 +247,9 @@ async def predict(
             input_batch = input_batch.to(self._device)
             try:
                 with torch.no_grad():
-                    outputs = self._model(**input_batch).logits
+                    outputs = self._model(**input_batch)
+                    if self.task != MLTask.text_embedding.value:
```

Comment:
good to add a comment why embedding task does not need logits

Reply:
got it, i refactored the code a bit to make this part about the `outputs` clearer!